### ISSUE TITLE
DXVK HUD Options not displaying changed options correctly when opened a second time

### ISSUE THREAD

**Issue by sonic2kk** [sonic2kk/steamtinkerlaunch#488]
The DXVK HUD options menu has a bug where selecting a menu item, closing the menu and opening the menu again will not display the updated menu option. However, `DXVK_HUD` is seemingly set correctly, as when a game launches it uses the correct options. 

Here's a set of steps to reproduce the behaviour, if there are already DXVK HUD variables selected you can skip to Step 4:
1. Open DXVK HUD options menu
2. Select a menu option e.g. `fps`
3. Press OK Button
4. Open DXVK HUD options menu again
5. Select a different menu option, e.g. `full`
6. Press OK button
7. Open DXVK HUD for a final time
8. The first option, e.g. `fps`, will still be selected
9. Press Cancel button
10. Launch game, DXVK HUD will display with the last selected menu item and not the one displayed in the menu the last time it was opened. In this example, it will use `full`, even though `fps` was displayed the last time the menu was open, which is correct behaviour but is a visual bug in the menus

I spent a while debugging this and the behaviour is a bit strange internally too. Basically when the menu is opened in Step 7, the `DXVK_HUD` variable is actually set to `fps`, so the logic for setting the selected menu items based on the `DXVK_HUD` variable *is* correct. It's the variable itself that's wrong. And it's wrong as far up as before the call to `DxvkHudPick`. Putting a log of `DXVK_HUD` before this call reveals that it's set incorrectly before the function is called! But putting a log after this call reveals that when changing the select HUD item, it's set correctly outside of this function. Here's an example:

```sh
...
# First open
DXVK_HUD is full before DxvkHudPick call
DxvkHudPick call
DXVK_HUD is fps after DxvkHudPick call

# Second open
DXVK_HUD is full before DxvkHudPick call
...
```

So basically the variable is set correctly when leaving the menu, but when you open the menu again, before the call to the menu is made the variable is wrong again. Perhaps not much can be done about this (maybe a similar issue to what was discussed in #440?) but it is kind of jarring if you want to go in and edit your DXVK HUD arguments again :sweat_smile:

**Comment by frostworx** [comment]
Thanks for reporting, there is definitely something wrong with that option.
Initially I thought that the variables are simply not updated, because the "in settings" menu button launches a 2nd steamtinkerlaunch instance, so the first instance doesn't get notified about any changes, but this is also reproducible
by directly using the undocumented `steamtinkerlaunch dxh` command line option. Changes don't seem to have any effect.
Probably easy to fix, but no idea when I find the time to look into it.
Will be away for ~10 days in 3 days and much to do until then.

**Comment by sonic2kk** [comment]
Thanks for looking into it! Enjoy the time away 😄

**Comment by frostworx** [comment]
Hah, seems like only the initial loading of the game config file is missing in the `DxvkHudPick` function.

edit: whoops, sorry, didn't mean to update the comment before writing, that I'll fix it now :)

**Comment by frostworx** [comment]
should be fixed :)

**Comment by sonic2kk** [comment]
Seems like its fixed! Thank you :)

**Comment by frostworx** [comment]
Thanks for testing :)

**Comment by sonic2kk** [comment]
Hm, so now I'm encountering a problem where the `DXVK_HUD=0` change doesn't seem to be working. It also doesn't seem to be working on my branch for it which is slightly worrying, even though I verified it was writing out last night. Though there was some odd behaviour in general, when trying to test your fix I had to restart my computer after installing STL. And to get any extra `writelog` commands to display in the STL logs I had to restart my machine *again*.

So I think this is just a weird problem with my computer, but could you test to see if `DXVK_HUD=0` is actually writing out to the game config file? Just to be sure, though I don't know why the code wouldn't work since it worked last night and also logically I'm not sure why it wouldn't write out. Putting a log for `writelog "INFO" "HUD var is '$DXVK_HUD'"` under the updateConfigEntry for nothing selected outputs the variable correctly. I really feel like something weird has happened with my STL installation but just to make completely extra sure :)

**Comment by frostworx** [comment]
ah, this is easy to fix. give me a sec. sorry, that I haven't tested this before merging.

**Comment by frostworx** [comment]
above commit should do it :)

**Comment by sonic2kk** [comment]
Can confirm that fixed it, woohoo! Thank you!! I should've taken more care, though I was pretty sure this was working before I submitted the PR. Either way, it works now :tada:

**Comment by frostworx** [comment]
cool, thanks for confirmation :)
